controller/project/manager.py

```diff
@@ -219,12 +229,182 @@ def __get_crowd_label_is_data(project_id: str, is_id: str) -> Dict[str, str]:
 
 def __get_first_data_id(project_id: str, user_id: str, huddle_type: str) -> str:
     if huddle_type == enums.LinkTypes.DATA_SLICE.value:
-        slices = ds_manager.get_all(project_id, enums.SliceTypes.STATIC_DEFAULT)
+        slices = data_slice.get_all(project_id, enums.SliceTypes.STATIC_DEFAULT)
         if slices and len(slices) > 0:
             return slices[0].id
     elif huddle_type == enums.LinkTypes.HEURISTIC.value:
-        return information_source_manager.get_first_crowd_is_for_annotator(
-            project_id, user_id
-        )
+        return information_source.get_first_crowd_is_for_annotator(project_id, user_id)
     else:
         raise ValueError("invalid huddle type")
+
+
+# GATES
+projects_updating = set()
+projects_updating_lock = threading.Lock()
+
+
+def get_gates_integration_data(project_id: str) -> GatesIntegrationData:
+
+    project_item = project.get(project_id)
+    if not project_item:
+        raise GraphQLError("Project not found")
+
+    if project_id in projects_updating:
+        return GatesIntegrationData(status=enums.GatesIntegrationStatus.UPDATING.value)
```

Comment:
dont know how perfomrance intensive the missing calculation is but I would have returned them in the updating case as well since with them we might be able to give some king of progression (at a later point in time --> findings)

Reply:
added to findings